src/Domain/DomainHelpers.cpp

```diff
@@ -838,17 +866,31 @@ std::vector<std::array<size_t, 8>> corners_for_radially_layered_domains(
        {{c[2], c[0], c[6], c[4]}}}};  // Lower x
   std::vector<std::array<std::array<size_t, 4>, 6>> faces_in_each_layer{
       faces_layer_zero};
+  const size_t offset = number_of_layers * 8;
```

Comment:
The corner numbering in this function is done by splitting the blocks in the domain into two groups (the blocks on the -x and the blocks on the +x) and numbering them separately (and then identifying corners on the shared boundaries) - to make sure the separate numberings have no collisions, the numbering of the +x blocks must be done with an offset relative to the -x blocks, the offset being the largest corner number expected to be used for the -x blocks, so that the corner numbers used for the +x blocks are all larger than the numbers used for the -x blocks